sklearn/preprocessing/label.py

```diff
@@ -891,7 +895,9 @@ def transform(self, y):
         """
         check_is_fitted(self, 'classes_')
 
-        class_to_index = dict(zip(self.classes_, range(len(self.classes_))))
+        if self._indexed_classes_cache is None:
+            self._indexed_classes_cache = dict(zip(self.classes_, range(len(self.classes_))))
+        class_to_index = self._indexed_classes_cache
```

Comment:
No need for this local variable. Combine these two lines

Reply:
done